### PR TITLE
Add feature matrix to README

### PR DESCRIPTION
**Pull request by pmorie**
Adds a feature matrix to the README per Jan 23 working group meeting

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,14 +48,23 @@ higher-level APIs:
   workloads should be spread across different clusters similar to how a human
   operator would
 
-# Guides
+## Features
 
-## User Guide
+| Feature | Maturity |
+|---------|----------|
+| Generate Federation APIs without writing code | Alpha |
```
Comment (gyliu513) [inline README.md]: @pmorie would you please link each feature to its related document?

Comment (pmorie) [reply README.md]: @gyliu513 what type of document are you thinking?

Comment (gyliu513) [reply README.md]: Like:

RSP: https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md#replicaschedulingpreference
Multicluster DNS: https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/servicedns-with-externaldns.md
Multicluster Ingress: https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/ingressdns-with-externaldns.md
Generate Federation APIs without writing code: https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md#enabling-federation-of-an-api-type

Comment (pmorie) [reply README.md]: ah ok

**Comment by pmorie** [comment]
@gyliu513 updated, sorry for delay, PTAL

**Comment by poothia** [comment]
lgtm

**Comment by gyliu513** [comment]
/lgtm

Thanks @pmorie

**Pull request merged**